validator/src/lib.rs

```diff
@@ -20,6 +25,69 @@ pub mod cli;
 pub mod commands;
 pub mod dashboard;
 
+#[cfg(unix)]
+fn redirect_stderr(filename: &str) {
+    use std::os::unix::io::AsRawFd;
+    match OpenOptions::new().create(true).append(true).open(filename) {
+        Ok(file) => unsafe {
+            libc::dup2(file.as_raw_fd(), libc::STDERR_FILENO);
+        },
+        Err(err) => eprintln!("Unable to open {filename}: {err}"),
+    }
+}
+
+#[cfg_attr(not(unix), allow(unused_variables))]
+fn create_signal_handler_thread(
+    logfile: Option<String>,
+    validator_exit: Arc<std::sync::RwLock<Exit>>,
+) -> Option<JoinHandle<()>> {
+    #[cfg(unix)]
+    {
+        solana_logger::setup_with_default_filter();
+        if let Some(ref logfile) = logfile {
+            redirect_stderr(logfile);
+        }
+
+        use log::{info, warn};
+        let mut signals = Signals::new([SIGTERM, SIGUSR1]).unwrap_or_else(|err| {
+            eprintln!("Unable to register SIGUSR1 handler: {err:?}");
+            exit(1);
+        });
+
+        std::thread::Builder::new()
```

Comment:
i mentioned this elsewhere, but reiterating here for posterity, we should refactor signal handling in multiple ways. 
1) all a signal handler should be doing is setting a global/singleton that's observed elsewhere
2) there's no reason for a thread to sit around waiting to handle signals